app/src/main/java/com/example/moodtrackerproject/ui/notes/NotesAdapter.kt

```diff
@@ -0,0 +1,62 @@
+package com.example.moodtrackerproject.ui.notes
+
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.example.moodtrackerproject.R
+import com.example.moodtrackerproject.data.NoteBody
+import kotlinx.android.synthetic.main.note_item.view.*
+
+class NotesAdapter : RecyclerView.Adapter<NotesAdapter.NotesHolder>() {
+
+    private var listOfNotes = emptyList<NoteBody>()
+    private var isFavourite = false
+
+    class NotesHolder(view: View) : RecyclerView.ViewHolder(view) {
+        val noteDate: TextView = view.note_text_date_time
+        val noteTitle: TextView = view.note_title_item
+        val noteText: TextView = view.note_text_item
+    }
+
+    override fun onViewAttachedToWindow(holder: NotesHolder) {
+        holder.itemView.setOnClickListener {
+            NotesFragment.click(listOfNotes[holder.adapterPosition])
+        }
+        holder.itemView.note_star_button_unchecked.setOnClickListener {
+            isFavourite = !isFavourite
+            if (isFavourite) {
+                holder.itemView.note_star_button_unchecked.setImageResource(R.drawable.ic_note_star_checked)
+            }
+            if (!isFavourite) {
+                holder.itemView.note_star_button_unchecked.setImageResource(R.drawable.ic_note_star_unchecked)
+            }
+        }
+    }
+
+    override fun onViewDetachedFromWindow(holder: NotesHolder) {
+        holder.itemView.setOnClickListener(null)
+        super.onViewDetachedFromWindow(holder)
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NotesHolder {
+        val view = LayoutInflater.from(parent.context).inflate(R.layout.note_item, parent, false)
+        return NotesHolder(view)
+    }
+
+    override fun onBindViewHolder(holder: NotesHolder, position: Int) {
+        holder.noteDate.text = listOfNotes[position].date
+        holder.noteTitle.text = listOfNotes[position].title
+        holder.noteText.text = listOfNotes[position].text
+    }
+
+    override fun getItemCount(): Int = listOfNotes.size
+
+    fun setList(list: MutableList<NoteBody>) {
```

Comment:
Why do you pass MutableList ?

Reply:
I'm only human
I'm just a man
I do what I can
Don't put the blame on me
BUT i'm still having a problem with this mutable list

Reply:
![image](https://user-images.githubusercontent.com/87538962/148660445-8d62f4b6-3266-483b-9bd9-15c5d40575f4.png)